### ISSUE TITLE
A bunch of labels have disappeared after loading a game

### ISSUE THREAD

**Issue by terzag** [MehVahdJukaar/labels-multi#29]

**Comment by MehVahdJukaar** [comment]
Do you have optiifine?

**Comment by terzag** [comment]
Sorry, I made a mistake when posting the message, so it went through without the description. :/

So, a big bunch of labels disappeared (about 80 out of 200) and from what I can tell, it seems that the missing ones are part of the same chunks. I.e. I have several barrels which had labels, they start missing at a specific column and I think it's at the exact limit between two chunks.

I've tried to quit/reload, go to the Nether and back (as I saw another report with issues when changing dimensions) but no change.

Not using Optifine, using Fabric + Sodium. Minecraft 1.21.

**Comment by MehVahdJukaar** [comment]
Ah hm weird. Did you update the mod? Did they pop in item form? They are entities so maybe something happened to them that broke them

**Comment by MehVahdJukaar** [comment]
Weird that it would happen out of the blue. Those aren't very different from item frames and if there was a bug where they would become invalid they should at least pop in item form

**Comment by terzag** [comment]
Didn't update the mod, I downloaded and installed it a few days ago.

_But_ I just reloaded the game again and they're back. So, not sure what happened. Is there a log to check if that happens again?

**Comment by MehVahdJukaar** [comment]
They were invisible then. I was aware of such an issue with optiifine. I'll check again near chunk borders

**Comment by terzag** [comment]
It just happened again and I might have a bit more information.

First, they're not invisible, they really seem to be unloaded, as if I hit the block they're (supposed to be) on, I get the "breaking block animation" and the label doesn't pop as an item. I'm really sure I hit in the center of the block, not just outside of the label.

Secondly, it seems to happen when I save the game next to some of them: when I load it back, they are missing. The ones in the chunk in front of me don't load, others in the chunk I am in, or behind, or somewhere else, load fine. If I save elsewhere and load, the ones previously missing are back.

I'll try to circle down the issue a bit (to see if it happens in a specific zone or if I get the same behaviour for other labels in other chunks).

**Comment by MehVahdJukaar** [comment]
I had previous reports saying they don't render when changing dimensions or going far away but I was never able to replicate. Maybe it's some incompatibility or weird interaction

**Comment by terzag** [comment]
Yeah. From what I've tried currently, it seems to happen only with a specific group of chunks. Not sure of it's related to the label themselves (i.e. a specific block set as the image on one label that triggers a bug on all the ones in the same chunk) or to the chunks they're in.

So far, I've been able to make them disappear at will (if I save next to them, they always disappear) but I haven't found a coherent way of making them reappear.

**Comment by MehVahdJukaar** [comment]
Anything specific about those chunks?

**Comment by terzag** [comment]
From what I can tell, no. There's a slime chunk just next to them but the ones where I have the labels have nothing special AFAIK.

**Comment by soulhakr** [comment]
I'm experiencing this same issue, running Minecraft 1.19.2, Better Minecraft modpack version 1.18.5 plus several additions, on a server. I'm using Sodium + Iris, and Complementary Reimagined Shaders. I'm seeing the exact same behavior, though it seems the labels sometimes return after events where the shaders reload - changing dimensions, going to sleep, etc.

**Comment by soulhakr** [comment]
I am suspecting that we may be running afoul of Entity Culling

**Comment by terzag** [comment]
Do you think it could be related to the number of labels that have to be displayed? In the chunks with issues I have 74 labels (across two chunks: 46 + 28 labels) that disappear. On the other side, I previously had item frames instead of them and didn't have issues with them, so if the labels are treated in a similar way, I guess it shouldn't be the case.

**Comment by MehVahdJukaar** [comment]
Definitely not. Had reports of this happening with just one. Does the same issue appear if you teleport from way from them to exactly their position?

**Comment by terzag** [comment]
How would I teleport?

**Comment by MehVahdJukaar** [comment]
should ve fixed

**Comment by terzag** [comment]
Well, because of the issues I have reverted to the old items-in-frames setup. I've put back a few labels (in the same chunks that triggered the issue) after I saw the update and they seem to display fine but I'd have to re-setup labels for the whole storage system to confirm if it's really fixed. Unfortunately, I won't have time to do this in the next few days but I'll report as soon as possible, in a week or so.

**Comment by MehVahdJukaar** [comment]
No worries. Unfortunately due to how the fix was handled old frames might pop off but should work fine after that

**Comment by terzag** [comment]
So, I've just taken the time to redo my whole storage area with labels and... there's a new issue, that might or not be related.

I started Minecraft and loaded my game, the few labels that I had put back for testing popped off as you said, I put labels back everywhere and added items to them. Everything looked fine. I saved, reloaded and the labels were still there, even in the chunks that caused issues before, but... they're blank.

I put one item back on a single label, saved again, reloaded, blank again.

I've tried the old trick that made them reappear when they vanished but no change, so that might be a completely different issue.

Well, at least the labels don't disappear anymore. :)

I've put the latest log there just in case there's something obvious: [https://pastebin.com/dmfCR0yH](https://pastebin.com/dmfCR0yH)
Also, recently, I've added some big mods (the kind that adds custom stuff to the game) so I wouldn't cross out a conflict with another mod.

**Comment by MehVahdJukaar** [comment]
darn. i tried replicating this and as usual i couldnt... try using /data get on them before and after you reload. it should show the missing data. also are you on fabric 1.20?

**Comment by terzag** [comment]
Using Fabric 1.20.1.

I don't really understand what you're asking me. Where/how do I use this /data command?

**Comment by terzag** [comment]
OK, I know where the issue comes from: shaders. Not sure if it's the one I use specifically (ComplementaryReimagined) as I don't have others to test.I'll try to find some.

Basically: I load my game, the labels are blank, I disable the shaders, I save and quit, I reload my game (with shaders still disabled), the labels are back −all of them.

**Comment by jacob99200** [comment]
> darn. i tried replicating this and as usual i couldnt... try using /data get on them before and after you reload. it should show the missing data. also are you on fabric 1.20?

I was having the same issue on a server using this mod

Relogging would cause ALL labels to go blank

Checking the data tho, the item data tho, it seems its still keeping the data just fine

Before Relog
![image](https://github.com/MehVahdJukaar/labels-multi/assets/6218816/fc093cfb-b0a8-43b0-8a66-1a10893644fa)
After Relog
![image](https://github.com/MehVahdJukaar/labels-multi/assets/6218816/733159f4-79fa-449b-9222-8fee082b9c7d)

This is with other mods tho, I will attempt this again only using the labels mod

**Comment by jacob99200** [comment]
At first I thought it was an issue with faux, but nope

This happens with only fabric API, labels, and moonlight installed

Minecraft Version: 1.20.1
Fabric Version: 0.14.22
Fabric API Version: 0.87.0+1.20.1
Labels Version: 1.20-1.16
Moonlight Version: 1.20-2.8.12

This time tested on singleplayer world:
Before Relog:
![image](https://github.com/MehVahdJukaar/labels-multi/assets/6218816/a2d81470-8f4b-4fdf-86d8-f4e948451dbc)
After Relog:
![image](https://github.com/MehVahdJukaar/labels-multi/assets/6218816/b82b0395-40d9-4dfb-a309-c50f21a51a1e)

**Comment by terzag** [comment]
There's something I didn't specify in my previous message: I tried with a new setup with only Fabric + Moonlight + Labels to see if the issue happened with the most basic configuration and race a mod conflict. It didn't. That's when I figured that one difference was the shaders and I tried disabled them on my usual configuration.

Now, seeing that @jacob99200 has the issue with a similar very basic setup, I checked again with mine and it seems that the problem actually happens too but does randomly: I start Minecraft, load the game, the label I've placed is blank. If I quit/reload, it reappears or not. I've tried to save/quit + reload several times in a row, it seems to be completely random: sometimes the label will display, sometimes not.

**Comment by KavisKryptic** [comment]
Just bumping this issue, since I'm running into it as well

**Comment by plasmagaming8** [comment]
Also bumping this issue! I'm having the same problem using Quilt loader 1.20.1 + sodium; no shaders

**Comment by rikanotank1** [comment]
> There's something I didn't specify in my previous message: I tried with a new setup with only Fabric + Moonlight + Labels to see if the issue happened with the most basic configuration and race a mod conflict. It didn't. That's when I figured that one difference was the shaders and I tried disabled them on my usual configuration.
> 
> Now, seeing that @jacob99200 has the issue with a similar very basic setup, I checked again with mine and it seems that the problem actually happens too but does randomly: I start Minecraft, load the game, the label I've placed is blank. If I quit/reload, it reappears or not. I've tried to save/quit + reload several times in a row, it seems to be completely random: sometimes the label will display, sometimes not.

Same issue here, hope dev fix labels they becoming blank after relog in game.

**Comment by MehVahdJukaar** [comment]
fixed
